### PR TITLE
Core/Objects: increase sight range of objects & correct general default visibility distance

### PR DESCRIPTION
**Pull request by Kittnz**
This can be used for anything really.
Note: This does not make the object active.

Some info on range
https://trinitycore.atlassian.net/wiki/spaces/tc/pages/74950924/Lua+Scene+Script+Documentation#LuaSceneScriptDocumentation-ActorAOIRange

#21111 #21681

### CODE REVIEW

File: src/server/game/Entities/GameObject/GameObjectData.h [hunk 1]
```diff
@@ -571,6 +571,23 @@ struct GameObjectTemplate
         }
     }
 
+    bool IsLargeGameObject() const
+    {
+        switch (type)
+        {
+        case GAMEOBJECT_TYPE_BUTTON: return button.large != 0;
```
Comment (Ovahlord) [inline src/server/game/Entities/GameObject/GameObjectData.h]: tabs are missing

File: src/server/game/Entities/GameObject/GameObjectData.h [hunk 2]
```diff
@@ -575,14 +575,14 @@ struct GameObjectTemplate
...
         {
-        case GAMEOBJECT_TYPE_BUTTON: return button.large != 0;
-        case GAMEOBJECT_TYPE_QUESTGIVER: return questgiver.large != 0;
-        case GAMEOBJECT_TYPE_GENERIC: return _generic.large != 0;
-        case GAMEOBJECT_TYPE_TRAP: return trap.large != 0;
-        case GAMEOBJECT_TYPE_SPELL_FOCUS: return spellFocus.large != 0;
-        case GAMEOBJECT_TYPE_GOOBER: return goober.large != 0;
-        case GAMEOBJECT_TYPE_SPELLCASTER: return spellcaster.large != 0;
-        case GAMEOBJECT_TYPE_CAPTURE_POINT: return capturePoint.large != 0;
+        case GAMEOBJECT_TYPE_BUTTON:            return button.large != 0;
```
Comment (Ovahlord) [inline src/server/game/Entities/GameObject/GameObjectData.h]: tabs are partially still missing :D TC standard is using one additional tab before the case starts. See the other helpers in that struct

**Comment by illfated** [comment]
Cosmetic detail in the PR title: "it's" means "it is", whereas you probably meant to write "its" which is the possessive form, but only in the singular form.
Then again, it would be preferable to use the plural form in the title, such as:
**Core/Objects: Have all large game objects' sight range increased & correct general default visibility distance**
(the possessive form when a noun ends in a plural S is the apostrophe - "objects' ")

File: src/server/game/Entities/Object/ObjectDefines.h [hunk 3]
```diff
@@ -28,7 +28,12 @@
 #define TRADE_DISTANCE              11.11f
 #define MAX_VISIBILITY_DISTANCE     SIZE_OF_GRIDS           // max distance for visible objects
 #define SIGHT_RANGE_UNIT            50.0f
-#define DEFAULT_VISIBILITY_DISTANCE 90.0f                   // default visible distance, 90 yards on continents
+#define SIGHT_GIGANTIC              400.0f
+#define SIGHT_LARGE                 200.0f
```
Comment (Faq) [inline src/server/game/Entities/Object/ObjectDefines.h]: why exact 200?

Comment (Kittnz) [reply src/server/game/Entities/Object/ObjectDefines.h]: https://trinitycore.atlassian.net/wiki/spaces/tc/pages/74950924/Lua+Scene+Script+Documentation#LuaSceneScriptDocumentation-ActorAOIRange

Comment (Kittnz) [reply src/server/game/Entities/Object/ObjectDefines.h]: From official sources.

**Comment by xvwyh** [comment]
Notice how "large" is the same property that got renamed to "GiganticAOI" in later expansions. That raises a question: did they just decide to increase visibility distance of all existing GOs from 200 to 400, or perhaps in WotLK times "large" actually meant 400, and "gigantic" wasn't a thing at all?

https://www.youtube.com/watch?v=P5Mbzodoxmo
Here's a WotLK video in which you can quite nicely judge what's supposed to be the distance of "large" GOs. Now how we approach this depends on whether we use 2D or 3D distance. Right after the player is launched from the cannon you can see ONE of the THREE cone-shaped tents visible at Darkmoon Faire grounds.
darkmoonfaire
That one tent seems to be at a distance of around 245 yards in 2D (275 in 3D). The other two are at a distance of about 270 (305 in 3D) and 305 (340 in 3D). I never had a reason to think blizzard uses 3D distance for visibility, correct me on this if I'm wrong. So assuming it's 2D, the "large" distance seems to actually be, I'd say, 250 yards? Or 300, on the off chance they did use 3D distance.

More videos proving the same:
https://youtu.be/ywEpt0u0Qa8?t=12
https://youtu.be/h3hfgBXT7Oc?t=8
https://youtu.be/n5kBOJogtnU?t=8

Sadly this doesn't give us a chance to prove anything about whether post-WotLK distances are indeed 200 and 400. You'd think why would they feel the need to decrease "large" from 250 to 200. But that would, however, explain why they replaced "large" GO property with "GiganticAOI": to ensure that old GOs would still be visible at the distance designers might have intended.

**Comment by Ovahlord** [comment]
Gigantic should be correct. I told @Kittnz  about that already in IRC. I have been porting the AOI naming fields from master to 4.3.4 since it contains Gigantic and Infinite AOI data as well (Gilneas has some weird acting doors in sniff which confirm this) and I have also noticed that on the master branch all large fields have been replaced by Gigantic so ye. Scourgeholme in Icecrown also have a plaguespreader and a watch tower that needs to be visible via gigantic aoi. And I doubt that Blizz decided midway that they want to change it to Gigantic since 200 is clearly too less range for certain gameobjects.
wowscrnshot_122818_124543

For 4.3.4 I have quite alot of gameobjects that confirm that the large field should be Gigantic like the magic barrier at the Lost City of the Tol'Vir.
wowscrnshot_122818_124737

**Comment by xvwyh** [comment]
@Ovahlord, you don't have to "prove" that GO property values changed, because they are sniffed. You don't have to "prove" that GO property definitions changed, because all the info about them (seen in the comments of GameObjectData.h) is contained within the exe even to this day. In Cata, yes, they already switched "large" property to "Gigantic AOI", so you should be looking at #22892 for anything Cata-related. This PR, however, is for WotLK, in which said property was still "large", so visibility distance could've been different from what it became on later expansions.

File: src/server/game/Globals/ObjectMgr.cpp [hunk 4]
```diff
@@ -1101,6 +1110,13 @@ void ObjectMgr::LoadCreatureAddons()
             creatureAddon.emote = 0;
         }
 
+        if (creatureAddon.visibilityDistanceType >= VisibilityDistanceType::Max)
+        {
+            TC_LOG_ERROR("sql.sql", "Creature (GUID: " UI64FMTD ") has invalid visibilityDistanceType (%u) defined in `creature_addon`.",
+                guid, AsUnderlyingType(creatureAddon.visibilityDistanceType));
```
Comment (illfated) [inline src/server/game/Globals/ObjectMgr.cpp]: I hope you have got a plan for dealing with this feedback from travis-ci :
```cpp
/home/travis/build/TrinityCore/TrinityCore/src/server/game/Globals/ObjectMgr.cpp:1116:17:
 fatal error: 
      format specifies type 'unsigned long' but the argument has type
      'ObjectGuid::LowType' (aka 'unsigned int') [-Wformat]
                guid, AsUnderlyingType(creatureAddon.visibilityDistanceType));
~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment (Kittnz) [reply src/server/game/Globals/ObjectMgr.cpp]: fixed

**Comment by illfated** [comment]
Sorry for mentioning this again, but I think the title could be improved / simplified
(just like your other PR to the `master` branch):
**Core/Objects: increase sight range of objects & correct general default visibility distance**

(Feel free to ignore this if you find it redundant.)

**Review by ghost — APPROVED** [review]

File: src/server/game/Globals/ObjectMgr.cpp (hunk 4 above)
Comment (illfated) [reply src/server/game/Globals/ObjectMgr.cpp]: (you may mark this conversation as resolved if you want)

**Comment by qyh214** [comment]
I have test and cool but maybe missing some bosses,for example:
```sql
-- ICC Lord Marrowgar 
-- ICC Lord Marrowgar 10
INSERT INTO `creature_template_addon` (`entry`, `visibilityDistanceType`) VALUES ('36612', '5'); 
-- ICC Lord Marrowgar 10h
INSERT INTO `creature_template_addon` (`entry`, `visibilityDistanceType`) VALUES ('37957', '5'); 
-- ICC Lord Marrowgar 25
INSERT INTO `creature_template_addon` (`entry`, `visibilityDistanceType`) VALUES ('37958', '5'); 
-- ICC Lord Marrowgar 25h
INSERT INTO `creature_template_addon` (`entry`, `visibilityDistanceType`) VALUES ('37959', '5'); 
```

**Review by qyh214 — APPROVED** [review]

**Comment by Kittnz** [comment]
If you want to add it, add it to creature_addon. The entrys that re in creature_template_addon are sniffed. Shouldn't be changed unless we have more details.

**Pull request merged**